### PR TITLE
New Release 1.10

### PR DESCRIPTION
**Pull request by Raycoms**
It's been a while since we had the last release, this beta runs for quite a while now quite smoothly.
I think it's time to release it for real.

I will let this pr open for a few days if there are no objections I will merge it.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHomeBuilding.java [hunk 1]
```diff
@@ -1,24 +1,146 @@
...
+
+        refreshView();
+    }
+
+    /**
+     * Action when an assign button is clicked.
+     *
+     * @param button the clicked button.
+     */
+    private void assignClicked(@NotNull final Button button)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHomeBuilding.java]: MAJOR Remove this unused method parameter "button". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 2]
```diff
@@ -869,18 +848,22 @@ public final void markDirty()
      * register a block and position.
      *
      * @param block to be registered
-     * @param pos of the block
+     * @param pos   of the block
+     * @param world the world to place the tileentity in
      */
-    public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos)
+    public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos, @NotNull final World world)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this unused method parameter "block". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 3]
```diff
@@ -1750,42 +1752,44 @@ private void hitBlockWithToolInHand(@Nullable final BlockPos blockPos, final boo
...
-            worldObj.setBlockToAir(blockPos);
+                    block.getSoundType(blockState, CompatibilityUtils.getWorld(this), blockPos, this).getVolume(),
+                    block.getSoundType(blockState, CompatibilityUtils.getWorld(this), blockPos, this).getPitch());
+            CompatibilityUtils.getWorld(this).setBlockToAir(blockPos);
 
             damageItemInHand(1);
         }
         else
         {
             //todo: might remove this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 4]
```diff
@@ -1750,42 +1752,44 @@ private void hitBlockWithToolInHand(@Nullable final BlockPos blockPos, final boo
...
-            if (!worldObj.isRemote)
+            if (!CompatibilityUtils.getWorld(this).isRemote)
             {
                 MineColonies.getNetwork().sendToAllAround(
                         //todo: correct side
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 5]
```diff
@@ -1676,18 +1678,18 @@ private void tryPickupEntityItem(@NotNull final EntityItem entityItem)
...
-            if (InventoryUtils.isItemStackEmpty(resultStack) || InventoryUtils.compareItemStacksIgnoreStackSize(itemStack, resultStack))
+            final int resultingStackSize = ItemStackUtils.getSize(resultStack);
+            if (ItemStackUtils.isEmpty(resultStack) || ItemStackUtils.compareItemStacksIgnoreStackSize(itemStack, resultStack))
             {
-                this.worldObj.playSound((EntityPlayer) null,
+                CompatibilityUtils.getWorld(this).playSound((EntityPlayer) null,
                         this.getPosition(),
                         SoundEvents.ENTITY_ITEM_PICKUP,
                         SoundCategory.AMBIENT,
                         0.2F,
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: MAJOR Assign this magic number 0.2F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 6]
```diff
@@ -1676,18 +1678,18 @@ private void tryPickupEntityItem(@NotNull final EntityItem entityItem)
...
                         (float) ((this.rand.nextGaussian() * 0.7D + 1.0D) * 2.0D));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: MAJOR Assign this magic number 0.7D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 7]
```diff
@@ -1,50 +1,74 @@
...
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
+
+import static com.minecolonies.api.util.constant.ToolLevelConstants.TOOL_LEVEL_WOOD_OR_GOLD;
 
 /**
  * The lumberjacks building.
  */
 public class BuildingLumberjack extends AbstractBuildingWorker
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java]: MINOR Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 8]
```diff
@@ -157,8 +248,12 @@ public AbstractJob createJob(final CitizenData citizen)
     public static class View extends AbstractBuildingWorker.View
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java]: MAJOR Reduce this class from 70 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java [hunk 9]
```diff
@@ -800,7 +890,7 @@ public void updateElement(final int index, @NotNull final Pane rowPane)
                 }
 
                 rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(name);
-                final boolean isTriggered = townHall.getColony().getPermissions().hasPermission(Permissions.Rank.valueOf(actionsList.getParent().getID().toUpperCase()), action);
+                final boolean isTriggered = townHall.getColony().getPermissions().hasPermission(Rank.valueOf(actionsList.getParent().getID().toUpperCase()), action);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java]: MINOR Define the locale to be used in this String operation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java [hunk 10]
```diff
@@ -13,12 +14,19 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import static com.minecolonies.api.util.constant.ToolLevelConstants.*;
+
 /**
  * The abstract class for each worker building.
  */
 public abstract class AbstractBuildingWorker extends AbstractBuildingHut
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java]: MINOR Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java [hunk 11]
```diff
@@ -237,6 +249,15 @@ public boolean hasResident(final CitizenData citizen)
...
+     */
+    public void checkIfFoodNeeded()
+    {
+        setFoodNeeded(residents.stream().filter(resident -> resident.getSaturation() < EntityCitizen.HIGH_SATURATION).findFirst().isPresent());
+    }
+
+    /**
      * The view of the citizen hut.
      */
     public static class View extends AbstractBuildingHut.View
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java]: MAJOR Reduce this class from 38 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java [hunk 12]
```diff
@@ -103,7 +103,7 @@ public IBlockState getSolidSubstitution(BlockPos ignored)
     private static boolean isOre(final Block block)
     {
         //TODO make this more sophisticated
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 13]
```diff
@@ -504,56 +493,70 @@ else if (itemsToDeliver.isEmpty())
      * Gathers only one stack of the item.
      *
      * @param buildingToDeliver building to deliver to.
-     * @return true if continue, false if not succuesful
      */
-    private boolean gatherItems(@NotNull final AbstractBuilding buildingToDeliver, @NotNull final BlockPos position)
+    private AIState gatherItems(@NotNull final AbstractBuilding buildingToDeliver, @NotNull final BlockPos position)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: MAJOR Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/blockout/controls/TextField.java [hunk 14]
```diff
@@ -315,10 +317,10 @@ public void onFocus()
         cursorBlinkCounter = 0;
     }
 
-    @Override
     /**
      * Draw itself at positions mx and my.
      */
+    @Override
     protected void drawSelf(final int mx, final int my)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/blockout/controls/TextField.java]: MAJOR The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
MAJOR Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java [hunk 15]
```diff
@@ -108,11 +109,11 @@ public EntityAIMeleeGuard(@NotNull final JobGuard job)
     @Override
     protected AIState searchTarget()
     {
-        if (checkForWeapon())
+        if (checkForToolOrWeapon(ToolType.SWORD))
         {
             return AIState.GUARD_SEARCH_TARGET;
         }
-        InventoryFunctions.matchFirstInProvider(worker, stack -> stack != null && Utils.doesItemServeAsWeapon(stack), worker::setHeldItem);
+        InventoryFunctions.matchFirstInProviderWithSimpleAction(worker, stack -> !ItemStackUtils.isEmpty(stack) && ItemStackUtils.doesItemServeAsWeapon(stack), worker::setHeldItem);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java]: MAJOR Split this 181 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/colony/ColonyView.java [hunk 16]
```diff
@@ -75,6 +85,11 @@
     private double overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyView.java]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java [hunk 17]
```diff
@@ -50,30 +50,47 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java]: MAJOR Reduce the number of returns of this method 9, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java [hunk 18]
```diff
@@ -1,21 +1,26 @@
...
 import com.minecolonies.coremod.colony.jobs.JobBuilder;
-import com.minecolonies.coremod.colony.workorders.WorkOrderBuildDecoration;
 import com.minecolonies.coremod.colony.workorders.WorkOrderBuild;
-import com.minecolonies.coremod.configuration.Configurations;
+import com.minecolonies.coremod.colony.workorders.WorkOrderBuildDecoration;
 import com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIStructure;
 import com.minecolonies.coremod.entity.ai.util.AIState;
 import com.minecolonies.coremod.entity.ai.util.AITarget;
-import com.minecolonies.coremod.util.*;
+import com.minecolonies.coremod.tileentities.TileEntityRack;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java]: MINOR Remove this unused import 'com.minecolonies.coremod.tileentities.TileEntityRack'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by Raycoms** [comment]
same thing here, Idk why we get those conflicts...

**Comment by Raycoms** [comment]
Same thing here @OrionDevelopment any idea why we get all those conflicts?

**Comment by Asherslab** [comment]
Interesting... btw once we figure out the Conflicts i'll go through and try and fix as many of these sonar and codacy stuff :smile:

**Pull request merged**